yarpcerrors/errors.go

```diff
@@ -26,6 +26,19 @@ import (
 	"fmt"
 )
 
+// Wrap returns a new Status that wraps the original error.
+//
+// The Code should never be CodeOK, if it is, this will return nil.
+func Wrap(code Code, err error) *Status {
```

Comment:
What do you think about renaming this method `FromErrorAndCode(code Code, err error) *Status`? One of the benefits is that it will be consistent with the naming of `func FromError(err error) *Status {` which does something almost similar.

Reply:
This actually makes much more sense, as FromError does exactly what I need, but with hardcoded code.

Will update the diff. Thanks!

Reply:
Looking more through the code it seems that `FromError` has a different semantics than `Wrap`:

- `FromError` extracts a `Status` from the error chain, which means that part of the error chain might be stripped.
- `Wrap` wraps the error chain with _the_ YARPC status specified, i.e. the consumer controls what status will be propagated up the stack (e.g. to other middleware).

For me, the best option is to either align the semantics or keep the name different. I would prefer the second option, as wrapping is what I am looking for.

Let me know what you think.